recipes-openxt/qemu-dm/files/0001-generic-xenstore-extensions.patch

```diff
@@ -123,6 +127,32 @@ Index: qemu-1.4.0/hw/xen_backend.c
 +}
 +
 +/**
++ * Un-register an existing xenstore node using the specified callback and opaque pointer.
++ * @returns 0 on success, -1 otherwise.
++ */
++int xenstore_remove_watch(const char *base, const char *node,
++                          xenstore_watch_cb_t cb, void *opaque)
++{
```

Comment:
Other XenStore helpers did not, so I assumed pointer validation was up to the caller. It made some sense to me since, in the current use-case, arguments are temporary static buffers allocated on the stack and passing cb/opaque as NULL is "safe" for add/remove helpers.
Adding a check in xenstore_update sounds like a good idea though, since the callback and opaque pointers are sscanf from the node without check.